src/core/device_constructor.jl

```diff
@@ -0,0 +1,18 @@
+function construct_device!(ps_m::CanonicalModel,
+                            device_model::DeviceModel,
+                            system_formulation::Type{S},
+                            system::PSY.PowerSystem,
+                            time_range::UnitRange{Int64};
+                            kwargs...) where {S <: PM.AbstractPowerFormulation}
+
+    construct_device!(ps_m,
+    device_model.device,
+    device_model.formulation,
+    system_formulation,
+    system,
+    time_range;
+    kwargs...)
+
+    return
+
+end
```

Comment:
This is a lone `construct_device!` function in the `src/core` folder. All the other functions are defined in `src/devices/device_constructors/*.jl`. Is there a technical reason for this organization? I get that this functions takes a higher set of input arguments and uses the attributes of the different objects to dispatch accordingly. But it does make it odd to read the code / understand where the methods are that will be dispatched. My suggestion would be to put all the `construct_device!` functions in the same file, separated by block comments. But I'm not too concerned about this one.

Reply:
I think that since there are going to be many `construct_device!` functions given the possible devices formulations then that file will get way to large. I think we can address the concerns on file organization later.